File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    var answer = '';
    
    [...s].forEach((c) => {
        let i_c = c.charCodeAt();
        for(let i=1; i<=index; i++){
            i_c++;
            
            if(i_c == 123)
                i_c = 97;
            
            while(skip.includes(String.fromCharCode(i_c))){
                i_c++;
                
                if(i_c == 123)
                    i_c = 97;
            }
        }
        
        answer += String.fromCharCode(i_c);
    });
    
    return answer;
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    set<char> set_s;
    
    for(int i=0; i<skip.length(); i++)
        set_s.insert(skip[i]);
    
    for(int i=0; i<s.length(); i++){
        char c = s[i];
        for(int j=1; j<=index; j++){
            c++;
            if(c == 'z'+1)
                c = 'a';
            
            while(set_s.find(c) != set_s.end()){
                c++;
                if(c == 'z'+1)
                    c = 'a';
            }
        }
        
        answer += c;
    }
    
    return answer;
}